src/reducers/operationsNew.ts

```diff
@@ -0,0 +1,46 @@
+import { Reducer } from 'redux';
+import { cloneDeep } from 'lodash';
+
+import { Uid, Mid, Id } from '../identifiers';
+import { OperationsAction, OperationsActionType } from '../actions';
+
+export enum OperationType {
+  REQUEST_INVITE = 'REQUEST_INVITE',
+  TRUST = 'TRUST'
+}
+export interface RequestInvitePayload {
+  full_name: string
+  to_uid: Uid
+  to_mid: Mid
+}
+export interface TrustPayload {
+  to_uid: Uid
+  to_mid: Mid
+}
+
+export interface APIOperationBase {
+  id: Id;
+  creator_mid: Mid;
+  creator_uid: Uid;
+}
+
+export type APIOperation = APIOperationBase & ({
+  op_code: OperationType.REQUEST_INVITE;
+  data: RequestInvitePayload;
+} | {
+  op_code: OperationType.TRUST;
+  data: TrustPayload
+})
+
+export type OperationsState = APIOperation[]
+
+export const reducer: Reducer<OperationsState> = (state = [], untypedAction) => {
+  const action = untypedAction as OperationsAction;
+  switch(action.type) {
+    case (OperationsActionType.SET_OPERATIONS): {
+      return cloneDeep(action.operations)
```

Comment:
From https://redux.js.org/faq/performance#performance-clone-state "Common Redux misconception: you need to deeply clone the state. Reality: if something inside doesn't change, keep its reference the same!"

If we absolutely need deep cloning here, let's add a comment explaining why and that it's probably a pattern that should not spread.